src/avx/avx_vector.rs

```diff
@@ -2207,38 +2141,38 @@ impl<T: AvxNum> AvxArrayMut<T> for &mut [Complex<T>] {
         self.as_mut_ptr()
     }
 }
-impl<T: AvxNum> AvxArrayMut<T> for &mut DoubleBuff<'_, T> {
+impl<'a, T: AvxNum> AvxArrayMut<T> for DoubleBuf<'a, T>
+where
+    Self: AvxArray<T>,
+    &'a mut [Complex<T>]: AvxArrayMut<T>,
+{
     #[inline(always)]
     unsafe fn store_complex(&mut self, data: T::VectorType, index: usize) {
-        debug_assert!(self.output.len() >= index + T::VectorType::COMPLEX_PER_VECTOR);
-        T::VectorType::store_complex(self.output.as_mut_ptr().add(index), data);
+        self.output.store_complex(data, index);
```

Comment:
It hadn't occurred to me to just forward the call like this, that's great. This reduces the duplication of the actual business logic too, making the copy/paste issue even less urgent.